thermostat_api_server.py

```diff
@@ -24,8 +24,9 @@
 api_server_address = os.environ['API_SERVER_ADDRESS']
 mqtt_address = os.environ['MQTT_SERVER']
 mqtt_port = int(os.environ['MQTT_PORT'])
-mqtt_username = os.environ['MQTT_USERNAME']
-mqtt_password = os.environ['MQTT_PASSWORD']
+if (os.environ['MQTT_USERNAME']):
```

Comment:
Need to check if the variable exists in the environment. This doesn't work as expected. This will do it:

`if 'MQTT_USERNAME' in os.environ:`

Reply:
Argh.  My PHP brain kicks in way too often.  :)  All fixed.